src/main/groovy/com/cloudbees/plugins/flow/FlowDSL.groovy

```diff
@@ -231,8 +244,7 @@ public class FlowDelegate {
             }
             if (paramValue instanceof Boolean) {
                 params.add(new BooleanParameterValue(paramName, (Boolean) paramValue))
-            }
-            else {
+            } else {
```

Comment:
why not check if:
. `paramValue` is an instance of `List` here,
. `paramName` is params / actions / values
then add the values to the correct place.